app/assets/javascripts/components/instructor_panel.js.jsx

```diff
@@ -7,6 +7,13 @@ var InstructorPanel = React.createClass({
       action: this.props.queuePop,
     };
   },
+  getQueuePinTopData: function () {
+    return {
+      title: 'Queue Pin Top',
+      className: this.buttonBaseClass + "huge primary " + (this.props.getUnpinnedRequests.length <= 0 ? "disabled" : ""),
```

Comment:
Since your `pinTopRequest` function returns `null` if there's nothing to pin, you could also check `this.props.pinTopRequest === null` here instead of having to pass down `getUnpinnedRequests`.